cvat-ui/src/containers/annotation-page/top-bar/top-bar.tsx

```diff
@@ -336,6 +407,34 @@ class AnnotationTopBarContainer extends React.PureComponent<Props> {
         copy(url);
     };
 
+    private checkUnsavedChanges(): void {
+        const {
+            jobInstance,
+            history,
+        } = this.props;
+
+        jobInstance.annotations.hasUnsavedChanges().then((unsaved: boolean) => {
+            if (unsaved) {
+                window.addEventListener('beforeunload', AnnotationTopBarContainer.beforeUnloadCallback);
```

Comment:
In this case can we have a lot of copies of event listeners for ``beforeunload`` to be setup?

Reply:
In case callback functions are the same object answer is no. But we can register few callbacks for `beforeunload` with different functions.

Reply:
Got it